llvm/lib/IRPrinter/IRPrintingPasses.cpp

```diff
@@ -75,11 +78,10 @@ PrintFunctionPass::PrintFunctionPass(raw_ostream &OS, const std::string &Banner)
 
 PreservedAnalyses PrintFunctionPass::run(Function &F,
                                          FunctionAnalysisManager &) {
-  // RemoveDIs: there's no textual representation of the DPValue debug-info,
-  // convert to dbg.values before writing out.
-  bool ShouldConvert = F.IsNewDbgInfoFormat;
-  if (ShouldConvert)
-    F.convertFromNewDbgValues();
+  // RemoveDIs: Regardless of the format we've processed this function in, use
+  // `WriteNewDbgInfoFormat` to determine which format we use to write it.
+  bool UsedNewDbgInfoFormat = F.IsNewDbgInfoFormat;
+  F.setIsNewDbgInfoFormat(WriteNewDbgInfoFormat);
```

Comment:
Could you make a scoped device for this idiom - for the set/reset rather than having them more loosely coupled like this?

Reply:
Done - I chose to keep the comments where they are and pass WriteNewDbgInfoFormat into the constructor rather than folding the comments and variable into the class, just because these are temporary insertions (until we turn off printing the old format) and the behaviour is non-obvious without the comment+variable imo - if you disagree then I can adjust it.